sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberAsyncClient.java

```diff
@@ -80,10 +94,8 @@
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<AreaCodes> getAllAreaCodes(
         String locationType, String countryCode, String phonePlanId, List<LocationOptionsQuery> locationOptions) {
-        LocationOptionsQueries locationOptionsQueries = new LocationOptionsQueries();
-        locationOptionsQueries.setLocationOptions(locationOptions);
-        return phoneNumberAdministrations.getAllAreaCodesAsync(
-            locationType, countryCode, phonePlanId, locationOptionsQueries);
+        return getAllAreaCodesWithResponse(locationType, countryCode, phonePlanId, locationOptions)
```

Comment:
Isn't it better to invoke the regular (without response) method and wrap in an exception.. this process of invoking getAllAreaCodesWithResponse and then trying to flatten doesn't look very clean?

Reply:
We can get the Entity model from the *WithResponse method, but we cannot get the Response information from the regular method (without response). So if we want to reuse the methods we need to start from the *WithResponse method.